### PR TITLE
Fold menu options

### PR DESCRIPTION
**Pull request by flycran**
# 增加折叠MenuOptions功能

意见来自https://github.com/ztjhz/FreeChatGPT/issues/79

1. 新增[src/assets/icons/ArrowBottom.tsx](https://github.com/ztjhz/FreeChatGPT/compare/main...flycran:FoldMenuOptions?expand=1#diff-5689f90f0f839816c4bbb75f75801b6625d69a0efd06f134964fc9624ae7d464)折叠图标
2. 新增[src/components/Menu/CollapseOptions.tsx](https://github.com/ztjhz/FreeChatGPT/compare/main...flycran:FoldMenuOptions?expand=1#diff-dfd51478a4f8b64d0ed5ce5d5ffaff1631fd65782df1ac53c01055e9480489c1)折叠按钮
3. 移除[src/components/Menu/ChatHistoryList.tsx](https://github.com/ztjhz/FreeChatGPT/compare/main...flycran:FoldMenuOptions?expand=1#diff-9ad0d964a38965d8a7f8b2fe8fd20c6c7ae5bcc019ab8b404d05dd0fe3324629)的分割线，将视觉内容移交由上述组件
4. 修改[src/components/Menu/MenuOptions/MenuOptions.tsx](https://github.com/ztjhz/FreeChatGPT/compare/main...flycran:FoldMenuOptions?expand=1#diff-89b07f127c3c6974e795ad195c3e5e7ab434e90307d73eede7f668d73e4e2bc7)内容以应用折叠组件并实现折叠效果

### CODE REVIEW

File: src/components/Menu/CollapseOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  .x5 {
+    padding: 0 5px;
+    border-radius: 3px;
+    cursor: pointer;
+  }
+  
+  .fold {
+    transform: rotate(180deg);
+  }
+`;
```
Comment (ztjhz) [inline src/components/Menu/CollapseOptions.tsx:28]: To ensure consistency throughout the project, please use tailwindcss

File: src/components/Menu/CollapseOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+const CollapseOptions = memo((props: { onSetFold: Function; fold: boolean }) => {
```
Comment (ztjhz) [inline src/components/Menu/CollapseOptions.tsx:30]: Please destructure the props.

File: src/components/Menu/MenuOptions/MenuOptions.tsx [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
 import ImportExportChat from '@components/ImportExportChat';
+import CollapseOptions from '@components/Menu/CollapseOptions';
 import SettingsMenu from '@components/SettingsMenu';
+import React, { useState } from 'react';
+import Api from './Api';
+import ClearConversation from './ClearConversation';
+import Me from './Me';
 
 const MenuOptions = () => {
+  const foldState = useState(false);
```
Comment (ztjhz) [inline src/components/Menu/MenuOptions/MenuOptions.tsx:11]: Please use array destructuring for the useState hook and include the type when declaring the state.

**Review by ztjhz — CHANGES_REQUESTED** [review]
There are some changes to be made. Note: this might conflict with the PR #78.

File: src/components/Menu/CollapseOptions.tsx (hunk 2 above)
Comment (flycran) [reply src/components/Menu/CollapseOptions.tsx:30]: 我不太理解你这一句的含义，能举例说明一下吗

File: src/components/Menu/CollapseOptions.tsx (hunk 1 above)
Comment (flycran) [reply src/components/Menu/CollapseOptions.tsx:28]:
```txs
<div className='flex items-center'>
      <div className='border-b border-white/20 flex-1'></div>
      <div className='bg-gray-900 hover:bg-gray-500/10 transition-colors duration-200 px-3 rounded-sm cursor-pointer'
           style={ { transform: props.fold ? 'rotate(180deg)' : '' } }
           onClick={ () => props.onSetFold(!props.fold) }>
        <ArrowBottom />
      </div>
      <div className='border-b border-white/20 flex-1'></div>
    </div>
```

File: src/components/Menu/MenuOptions/MenuOptions.tsx (hunk 3 above)
Comment (flycran) [reply src/components/Menu/MenuOptions/MenuOptions.tsx:11]:
```tsx
const MenuOptions = () => {
  const [ fold, setFold ] = useState<boolean>(false);
  return (
    <>
      <CollapseOptions onSetFold={ setFold } fold={ fold } />
      {
        fold || (
          <>
            <AboutMenu />
            <ClearConversation />
            <ImportExportChat />
            <Api />
            <SettingsMenu />
            <Me />
          </>
        )
      }
    </>
  );
};
```

**Comment by flycran** [comment]
image
这是目前的视觉效果

File: src/components/Menu/CollapseOptions.tsx (hunk 2 above)
Comment (flycran) [reply src/components/Menu/CollapseOptions.tsx:30]: 我为`fold`添加了持久化，现在它没有props了

**Comment by flycran** [comment]
大佬，麻烦审查以下

**Comment by flycran** [comment]
人嘞？

**Review by ztjhz — APPROVED** [review]
I have improvements.

**Pull request merged**